### DISCUSSION TITLE
Open Fuzzy Finder when typing `**f`

### DISCUSSION THREAD

**Discussion by primeapple** [PatrickF1/fzf.fish#295] [Q&A]
I'm an avid vim user and really really really don't like the `CTRL`, `META`, ... keys.

So I would love to have an solution where I would type something like

```
npm run test -- **f
```

and the File fuzzy finder would open immediately. If I pick a files (or multiple of them), they'll replace the `**f`. Fish has `anywhere` abbreviations now ( https://fishshell.com/docs/current/cmds/abbr.html ), maybe we could use this?

Do you guys have any ideas?

**Comment by PatrickF1** [comment]
Unfortunately, I don't think that'd work because even if you expanded **f to `(_fzf_search_directory)`, when you hit enter, `_fzf_search_directory` will execute in a different context where there is no command line to update. However, there could be some feature I'm missing so there's still hope.

**Reply by primeapple** [reply]
So I tried the following:
```fish
bind gf _fzf_search_directory
```

Now when I type the following sequence (`|` is the cursor in insert mode, `_` is the cursor in normal mode here):
```fish
# empty prompt
|
# entering "ls " (space is important here)
ls |
# pressing Escape to go to normal mode
ls_
# now when typing `gf` I get "ls" inside the fuzzy prompt
```
I can now pick my file, but the `ls` is gone. The only option to fix this is the following
```fish
# empty prompt
|
# entering "ls  " (two spaces are important here)
ls |
# pressing Escape to go to normal mode
ls_
# now when typing `gf` I get it perfectly working
```

Do you have an idea how to get this working without double space?
Maybe configure `_fzf_search_directory` so that it doesn't take word that I'm currently on as start of the prompt?

**Reply by PatrickF1** [reply]
it's probably because fzf uses `commandline --replace` when writing to the commandline. This was introduced a while back and I asked the community if it was okay to kill it but unfortunately, most people liked it so I kept it in.

**Reply by primeapple** [reply]
Is there an option to change this? It's probably either `append` or `insert`
